pkg/beholder/auth.go

```diff
@@ -0,0 +1,26 @@
+package beholder
+
+import (
+	"crypto/ed25519"
+	"fmt"
+)
+
+// authHeaderKey is the name of the header that the node authenticator will use to send the auth token
+var authHeaderKey = "X-Beholder-Node-Auth-Token"
+
+// authHeaderVersion is the version of the auth header format
+var authHeaderVersion = "1"
+
+// BuildAuthHeaders creates the auth header value to be included on requests.
+// The current format for the header is:
+//
+// <version>:<public_key_hex>:<signature_hex>
+//
+// where the byte value of <public_key_hex> is what's being signed
+func BuildAuthHeaders(privKey ed25519.PrivateKey, pubKey ed25519.PublicKey) map[string]string {
+	messageBytes := pubKey
```

Comment:
If we are passing the private key, can we just ask it for the public?
```suggestion
	messageBytes := privKey.Public()
```
I guess it adds a type cast...

Reply:
addressed here b969b66e6141c80565d51ecd64a8bb39d4c09811